api4/status.go

```diff
@@ -127,7 +127,7 @@ func updateUserCustomStatus(c *Context, w http.ResponseWriter, r *http.Request)
 	}
 
 	customStatus := model.CustomStatusFromJson(r.Body)
-	if customStatus == nil || (customStatus.Text == "" && customStatus.Emoji == "") {
+	if customStatus == nil || customStatus.Emoji == "" || !customStatus.IsDurationValid() || !customStatus.IsExpirationTimeValid() {
```

Comment:
Is it intentional that we're now requiring emojis to have an emoji now? Previously, we required at least one of emoji or text.

Reply:
Previously when we required one of emoji and text, it was wrong because not sending emoji in the request body was setting the emoji as an empty string. We can do two things - we can make emoji required or we can set speech_balloon when emoji is not sent in the body. So if you want to go with the latter one, I can make that change. What do you think?

Reply:
It is defined in the spec that if the emoji is not set then it defaults to the speech balloon

Reply:
@esethna Any thoughts about changing this? I'm not sure what the spec says here about the custom status requiring text or an emoji?

Reply:
@hmhealey sorry I must not be understanding. We don't want to make any changes to how it works today. Meaning:

- Users can set an emoji with no text
- Users can set text without an emoji (in this case the emoji defaults to the speech balloon)
- Users can set both text and emoji